src/diem/testing/miniwallet/app/app.py

```diff
@@ -171,7 +184,50 @@ def _send_external_payment_txn(self, txn: Transaction) -> None:
         else:
             self._start_external_payment_txn(txn)
 
-    def _start_external_payment_txn(self, txn: Transaction) -> None:
+    def _start_external_payment_txn(self, txn: Transaction) -> None:  # noqa: C901
+        if identifier.diem_id.is_diem_id(str(txn.payee)):
+            reference_id = str(uuid.uuid4())
+            while True:
+                try:
+                    self.store.create(
+                        ReferenceID,
+                        before_create=self._validate_unique_reference_id,
+                        account_id=txn.account_id,
+                        reference_id=reference_id,
+                    )
+                    break
+                except ValueError:
+                    reference_id = str(uuid.uuid4())
+                    continue
+            try:
+                vasp_identifier = identifier.diem_id.get_vasp_identifier_from_diem_id(str(txn.payee))
+                domain_map = self.diem_client.get_diem_id_domain_map()
+                payee_onchain_address = domain_map.get(vasp_identifier)
+                if payee_onchain_address is None:
+                    raise ValueError(f"Diem ID domain {vasp_identifier} was not found")
+                self.store.update(txn, reference_id=reference_id)
+                sender_diem_id = identifier.diem_id.create_diem_id(
+                    str(self.store.find(Account, id=txn.account_id).diem_id), self.diem_account.diem_id_domains()[0]
+                )
```

Comment:
we probably should pick diem_id_domain when create account (more close to real use case). So the Account#diem_id field should include diem_id_domain.
Otherwise the name is not accurate, as the first part of the diem id should be user identifier.